### DISCUSSION TITLE
Reduce number of skeleton rows displayed when loading

### DISCUSSION THREAD

**Discussion by amhassen97** [KevinVandy/material-react-table#385] [Q&A] [answered]
Hi 

How do you reduce the number of skeleton rows displayed (so as to reduce height taken on page whilst loading) whilst data is being fetched/loaded? Also, how do you at the same time set the number of rows per page displayed when the table is rendered (instead of the default 10)?

Much appreciated for any assistance.

**Comment by MarkusEllyton** [comment, marked as answer]
The pageSize-attribute of the pagination state controls both of these things. You can set it through either the `initialState` or `state` prop, depending on what you're trying to achieve. This example in the docs might also help https://www.material-react-table.com/docs/guides/pagination#manage-pagination-state

**Reply by amhassen97** [reply]
Perfect thank you so much!
